### DISCUSSION TITLE
Integrating Specific Device Faults (Stuck-at-High/Low Resistance) in HWA Training or Inference

### DISCUSSION THREAD

**Discussion by adnanrana88** [IBM/aihwkit#591] [Q&A]
Hello,

I have been working with the IBM Analog Hardware Acceleration Kit for simulating neural networks on analog devices and have a specific requirement. I am interested in integrating more detailed device fault simulations into the HWA training process as well during inference, specifically faults where devices are stuck at high or low resistance states. This is to mimic realistic hardware faults like stuck-at-open (high resistance) or stuck-at-Gmax (low resistance) conditions.

Could you provide guidance on how to incorporate these specific types of device fault simulations into the HWA training process? For instance, is there a way to modify the weight noise injection or other aspects of the HWA training to include these fault scenarios? An example of the type of fault I am interested in is:

Stuck-at-open: A device is always in a high resistance state.
Stuck-at-Gmax: A device is always in a low resistance state.
I appreciate any advice or examples you can provide on implementing this in the aihwkit framework.

Thank you!

**Comment by maljoras** [comment]
You could use the modifier settings: https://github.com/IBM/aihwkit/blob/master/src/aihwkit/simulator/parameters/inference.py#L136
